GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -694,6 +694,29 @@ public string DescribeRevision(GitRevision revision, int maxLength = 0)
             return description;
         }
 
+        /// <summary>
+        /// Get the (last) selected revision in the grid.
+        /// </summary>
+        /// <returns>The selected revisions or <see langword="null"/> if none selected.</returns>
+        public GitRevision? GetSelectedRevisionOrDefault()
+        {
+            foreach (DataGridViewRow row in _gridView.SelectedRows)
+            {
+                if (row.Index >= _gridView.RowCount)
+                {
+                    continue;
+                }
+
+                GitRevision? revision = GetRevision(row.Index);
+                if (revision is not null)
+                {
+                    return revision;
+                }
+            }
```

Comment:
Yes, no more need for LINQ here. Then perhaps
```suggestion
            int rowCount = _gridView.RowCount;
            foreach (DataGridViewRow row in _gridView.SelectedRows)
            {
                if (row.Index < rowCount && GetRevision(row.Index) is GitRevision revision)
                {
                    return revision;
                }
            }
```

Reply:
Adopting the pattern matching, keeping the two if. In this case I see the continue and return to be two separate parts, prefer to to just use one statement.

Reply:
`continue` is without need. A `foreach` with a single `return` on positive condition is much clearer.
And the two subconditions are related by `row.Index`.
You could insert a line break before the `&&`.

Reply:
I do not gree in this situation, but will change the code: I do not want you to stop reviewing.